### PR TITLE
Introduction of new dialog for easy access to basic device configuration details

### PR DESCRIPTION
**Pull request by chahr**
Implementation includes tabs for simple access to general, latency, devicechannel details in a non-modal device details dialog that can be invoked by right-click context menu action or double click on the respective row in entity table view.
To support displaying active end-to-end channel connections in tabs for devicechannel details, an avdecc helper class to process stream connections and dynamic mappings accordingly was implemented. (This is due to be extended in another task/ with further functionality to allow not only channel connected state determination but full channel based connection management)

### CODE REVIEW

File: src/deviceDetailsChannelTableModel.cpp [hunk 1]
```diff
@@ -0,0 +1,717 @@
...
+	virtual ~DeviceDetailsChannelTableModelPrivate();
+
+	int rowCount() const;
+	int columnCount() const;
+	QVariant data(QModelIndex const& index, int role) const;
+	bool setData(const QModelIndex& index, const QVariant& value, int role);
+	QVariant headerData(int section, Qt::Orientation orientation, int role) const;
+	Qt::ItemFlags flags(QModelIndex const& index) const;
+
+	void registerUiWidget(QTableView* ui);
```
Comment (epebble) [inline src/deviceDetailsChannelTableModel.cpp]: Seems strange to have something referring to the view in the model.

File: src/deviceDetailsChannelTableModel.cpp [hunk 2]
```diff
@@ -0,0 +1,717 @@
...
+		{
+			bool hasChanges = false;
+			if (_hasChangesMap.contains(_nodes.at(index.row()).audioClusterNode.descriptorIndex) && _hasChangesMap.value(_nodes.at(index.row()).audioClusterNode.descriptorIndex)->contains(DeviceDetailsChannelTableModelColumn::ChannelName))
+			{
+				hasChanges = true;
+			}
+			if (role == Qt::FontRole && hasChanges)
+			{
+				QFont font;
+				font.setItalic(true);
```
Comment (epebble) [inline src/deviceDetailsChannelTableModel.cpp]: Italic text for pending changes I s not part of Hive at this time, we must decide if this is something we would like to generalize at some point.

File: src/deviceDetailsChannelTableModel.cpp [hunk 3]
```diff
@@ -0,0 +1,717 @@
...
+
+	return {};
+}
+
+/**
+* Gets the flags of the table cells.
+*/
+Qt::ItemFlags DeviceDetailsChannelTableModelPrivate::flags(QModelIndex const& index) const
+{
+	if (index.column() == (int)DeviceDetailsChannelTableModelColumn::ChannelName)
```
Comment (epebble) [inline src/deviceDetailsChannelTableModel.cpp]: static_cast

File: src/mainWindow.cpp [hunk 4]
```diff
@@ -264,6 +265,24 @@ void MainWindow::connectSignals()
...
+	connect(controllerTableView, &QTableView::doubleClicked, this,
+		[this](QModelIndex const& index)
+		{
+			auto& manager = avdecc::ControllerManager::getInstance();
+			auto const entityID = _controllerModel->controlledEntityID(index);
+			auto controlledEntity = manager.getControlledEntity(entityID);
+
+			DeviceDetailsDialog* dialog = new DeviceDetailsDialog(this);
+			dialog->setControlledEntityID(entityID);
+			dialog->show();
```
[the remark this answers is not in the document]
Comment (chahr) [reply src/mainWindow.cpp]: We wanted to have the possibility to use multiple dialogs at the same time and therefor use the dialog as non-modal, so I would like to keep this as it is, if that is ok for you.

File: src/deviceDetailsChannelTableModel.cpp (hunk 1 above)
Comment (chahr) [reply src/deviceDetailsChannelTableModel.cpp]: I agree on this and we have removed that in the latest commits. It was really not needed after all.

File: src/deviceDetailsChannelTableModel.cpp (hunk 2 above)
Comment (chahr) [reply src/deviceDetailsChannelTableModel.cpp]: We have removed usage of italic fonts as change indication in the latest commits and use enabling/disabling the apply/revert buttons to indicate presence of local changes instead.

File: src/deviceDetailsChannelTableModel.cpp (hunk 3 above)
Comment (chahr) [reply src/deviceDetailsChannelTableModel.cpp]: Of course...

**Comment by christophe-calmejane** [comment]
Integrated in current Hive beta version

**Pull request closed without merging**